test/services/optIn/OptInPropositionExcelSpec.scala

```diff
@@ -0,0 +1,223 @@
+        case _ => throw new RuntimeException(s"Unexpected presented in View and Change value: $input")
+      }
+
+    def sentFromViewAndChange(input: String): String =
+      input.trim match {
+        case "N/A" => quote(" ")
+        case "CY" => quote("CY")
+        case "CY+1" => quote("CY+1")
+        case "" => quote(" ")
+        case _ => throw new RuntimeException(s"Unexpected sent from View and Change value: $input")
+      }
+
+    def quote(input: String): String = {
+      "\"" + input + "\""
+    }
+
+    println("""("Current Year", "Next Year", "Intent", "Valid", "Presented", "Sent", "Expected CY", "Expected NY"),""")
+
+    val tsvOptIn = Source.fromFile("test/resources/OptInScenarios.tsv")
+    val rowsWithoutHeaders = tsvOptIn.getLines().drop(2)
+
+    val tab = "  "
+
+    rowsWithoutHeaders.foreach(line => {
+      val cells = line.split(tab).take(8)
+
+      val lineString = Seq(
+        quote(parseItsaStatus(cells(0))),
+        quote(parseItsaStatus(cells(1))),
+        quote(parseCustomerIntent(cells(2))),
+        parseIsValid(cells(3)).toString,
+        presentedInViewAndChange(cells(4)).toString(),
+        sentFromViewAndChange(cells(5)),
+        quote(parseItsaStatus(cells(6))),
+        quote(parseItsaStatus(cells(7))),
+      ).mkString(", ")
+
+      val s = s"($lineString),"
+
+      println(s)
+    })
+
+    tsvOptIn.close()
+  }
+
+  val forYearEnd = 2021
+  val currentTaxYear: TaxYear = TaxYear.forYearEnd(forYearEnd)
+
+  private val scenarios = Table(
+    ("Current Year", "Next Year", "Intent", "Valid", "Presented", "Sent", "Expected CY", "Expected NY"),
+    ("M", "M", "CY", false, List(), " ", "M", "M"),
+    ("M", "M", "CY+1", false, List(), " ", "M", "M"),
+    ("M", " ", "CY", false, List(), " ", "M", " "),
+    ("M", " ", "CY+1", false, List(), " ", "M", " "),
+    ("M", "V", "CY", false, List(), " ", "M", "V"),
+    ("M", "V", "CY+1", false, List(), " ", "M", "V"),
+    ("M", "A", "CY", false, List("CY+1"), " ", "M", "A"),
+    ("M", "A", "CY+1", true, List("CY+1"), "CY+1", "M", "V"),
+    ("V", "M", "CY", false, List(), " ", "V", "M"),
+    ("V", "M", "CY+1", false, List(), " ", "V", "M"),
+    ("V", " ", "CY", false, List(), " ", "V", " "),
+    ("V", " ", "CY+1", false, List(), " ", "V", " "),
+    ("V", "V", "CY", false, List(), " ", "V", "V"),
+    ("V", "V", "CY+1", false, List(), " ", "V", "V"),
+    ("V", "A", "CY", false, List("CY+1"), " ", "V", "A"),
+    ("V", "A", "CY+1", true, List("CY+1"), "CY+1", "V", "V"),
+    ("A", "M", "CY", true, List("CY"), "CY", "V", "M"),
+    ("A", "M", "CY+1", false, List("CY"), " ", "A", "M"),
+    ("A", " ", "CY", true, List("CY", "CY+1"), "CY", "V", " "),
+    ("A", " ", "CY+1", true, List("CY", "CY+1"), "CY+1", "A", "V"),
+    ("A", "V", "CY", true, List("CY"), "CY", "V", "V"),
+    ("A", "V", "CY+1", false, List("CY"), " ", "A", "V"),
+    /* todo: check this line */
+    //("A", "A", "CY", true, List("CY", "CY+1"), "CY", "V", "V"),
```

Comment:
With this test, the implementation more is perhaps more geared to opt out where we send multi year to the backend whereas with Opt In we only send the one

Reply:
What's the change required here please?

Reply:
@ian-hmrc please clarify this comment, not clear on what's the required change here?
![Screenshot 2024-09-03 at 10 10 54](https://github.com/user-attachments/assets/2677974d-67be-4201-ad64-eff35c6104c5)

Reply:
With Opt-In, we initially had some ambiguity as to whether the front-end or the back-end would own the 'and onwards' functionality, i.e. whether we would just send the user's chosen tax year, or all the tax years from their choice and onwards.

This information is the best we have from the business on this decision, so, according to the sheet, it falls to the back-end to implement this. Let's align to this and make that row of the test pass by just sending the chosen year for now. Should we get a change of specification, then we can revisit this.

Reply:
done